nano/node/transport/transport.hpp

```diff
@@ -8,6 +8,27 @@
 
 namespace nano
 {
+class bandwidth_limiter final
+{
+public:
+	// initialize with rate 0 = unbounded
+	bandwidth_limiter (const size_t);
+	bool should_drop (const size_t &);
+	size_t get_rate ();
+
+private:
+	//last time rate was adjusted
+	std::chrono::steady_clock::time_point last_poll;
+	//trend rate over 20 poll periods
+	boost::circular_buffer<size_t> rate_buffer{ 20, 0 };
+	//limit bandwidth to
+	size_t limit;
```

Comment:
This can be const as it's not modified outside of the constructor member initializer list